src/quarterback/api/methods.ts

```diff
@@ -150,55 +156,75 @@ export function del<T>(
     defaultError
   )
 }
-
 export async function listen<T>(
   path: string,
-  listener: (event: EventSourceMessage) => void,
-  authToken: string,
-  defaultError?: string,
-  headers: Record<string, string> = {
-    ...DEFAULT_HEADERS,
-    ...getAuthHeader(authToken),
-  }
+  listener: (event: MessageEvent) => void,
+  projectID: string,
+  manuscriptID: string,
+  authToken: string
 ) {
-  await fetchEventSource(`${config.api.url}/${path}`, {
-    onmessage: listener,
-    headers: headers,
-    async onopen(response) {
-      if (
-        response.ok &&
-        response.headers.get('content-type') === 'text/event-stream'
-      ) {
-        if (response.headers.get('Transform-Version') !== getVersion()) {
-          console.warn(
-            `Warning! Manuscripts-transform (Frontend: ${getVersion()}) version is different on backend (${response.headers.get(
-              'Transform-Version'
-            )})`
-          )
-        }
-        console.log('EventSource Connection Opened Ok')
-        return
-      } else if (
-        response.status >= 400 &&
-        response.status < 500 &&
-        response.status !== 429
-      ) {
-        // client-side errors are usually non-retriable:
-        console.error(
-          'EventSource connection error with status: ' + response.status
-        )
-      } else {
-        console.error(
-          'EventSource connection error with status: ' + response.status
-        )
-      }
-    },
-    onclose() {
-      // if the server closes the connection unexpectedly, retry:
-      console.log('EventSource connection closed')
-    },
-    onerror(err) {
-      console.log('EventSource connection error: ' + err)
-    },
-  })
+  const reconnectDelay = 1500
```

Comment:
I wonder why it's 1500? It seem to me to be a good balanced value but I'm curious how it was produced.

Reply:
I just found it to be kind of  balanced during testing (1 second felt too fast), i tried multiple things, one of the options were to have an exponential backoff with some sort of limit, but it felt like if we want something like that we'll need some sort of a popup to the users to let them know we're attempting a reconnect. I'm not sure if you have any thoughts on this.

Reply:
Ok, thanks for the info